GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs

```diff
@@ -633,6 +647,26 @@ private void listView1_MouseClick(object sender, MouseEventArgs e)
             }
         }
 
+        private void TextBoxSearch_TextChanged(object sender, EventArgs e)
+        {
+            ShowRecentRepositories();
```

Comment:
This is a very handed way of performing the filter. Each time the user will type a character we will be reloading the configuration file, deserializing the list of repositories, recalculating tile sizes and render items.

The filtering must be performed on `recentRepositories` and `favouriteRepositories` collections that are loaded in `ShowRecentRepositories`, adding or removing tiles from the listview as necessary.

Reply:
I agree, but one question: The `SplitRcentRepos` may trim the length of the repo list, correct? So would it make sense to keep that list in memory (before the split) and apply the filter on that? This way the search will also find items which are not currently shown in the UI. Or would you expect to only filter the items currently visible?

Reply:
That's a very good Q!
I suppose I could see an expectation to get a result searching for "project" whilst having a repo displayed as `c:\myrepos\so....roject.git` (with the full name `c:\myrepos\somecompany\projects\coolproject.git`). But at the same time I can see why this search result is totally unexpected too.
Personally I'd go for a filter to what's being displayed. That is, if there is not "project" word is visible, then nothing is returned.

@gitextensions/git-extensions-team do you have opnions?

Reply:
If I understand the question: I would expect to search the full name, not just what happens to be displayed.

Reply:
@RussKie Maybe I maybe formulated the question badly. I meant the number of items shown (defined by `maxRecentRepositories`) might be less than the number of repos known to git extensions.
As I user I would expect the search to search for "any repo known to git extensions matching the patch" rather than "search in the couple of items shown in the UI".
Therefore the search needs to cache the raw list returned by `ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.LoadRecentHistoryAsync());`.
Do you agree?

Reply:
> RepositoryHistoryManager.Locals.LoadRecentHistoryAsync()

We trim the history size upon save, so the list you get is all we have:
https://github.com/gitextensions/gitextensions/blob/a32fe7a6b79cb3af43fed302a7b548687156d720/GitCommands/UserRepositoryHistory/LocalRepositoryManager.cs#L292-L313

Reply:
Ah I see, thanks for clarifying. So for now I just cache the result from `LoadRecentHistoryAsync()` to avoid loading the data from disk all the time. The search is still applied on the repo list itself.